packages/wds-event-tool/src/main/kotlin/org/webdevandsausages/events/controller/PatchEvent.kt

```diff
@@ -0,0 +1,48 @@
+package org.webdevandsausages.events.controller
+
+import arrow.core.Either
+import org.http4k.contract.ContractRoute
+import org.http4k.contract.div
+import org.http4k.contract.meta
+import org.http4k.core.*
+import org.http4k.lens.string
+import org.webdevandsausages.events.Router
+import org.webdevandsausages.events.dto.EventUpdateInDto
+import org.webdevandsausages.events.error.toResponse
+import org.webdevandsausages.events.service.GetEventByIdService
+import org.webdevandsausages.events.service.UpdateEventService
+
+object PatchEvent {
+    private val eventUpdateRequestLens =
+        Body.string(ContentType.APPLICATION_JSON).map { json -> EventUpdateInDto.from(json) }.toLens()
```

Comment:
Was this the thingie you had in mind? Now we do the jackson object mapping in the Lens and we can also handle the case of invalid json input

Reply:
yep, cool this actually works...wasn't sure